src/platform/packages/shared/kbn-connector-specs/src/lib/get_schema_for_auth_type.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { z } from '@kbn/zod/v4';
+import type { NormalizedAuthType } from '../connector_spec';
+
+export const AUTH_TYPE_DISCRIMINATOR = 'authType';
+interface AuthTypeOverride {
+  defaults: Record<string, unknown> | undefined;
+}
+
+export const getSchemaForAuthType = (
+  id: string,
+  authType: NormalizedAuthType,
+  { defaults }: AuthTypeOverride
+) => {
```

Comment:
What is the return type here? There is already a type `ActionTypeSecrets` which is just `Record<string, unknown>`.

It would be useful to have something like `{ authType:  AuthTypeDiscriminators } | Record<string, unknown>`, then we could pass it around whenever `secrets` are necessary.

It would be useful to have this somewhere.